src/lib/spinel/radio_spinel.cpp

```diff
@@ -2150,7 +2148,7 @@ void RadioSpinel::RestoreProperties(void)
 
     if (mMacFrameCounterSet)
     {
```

Comment:
do we still need this flag?

Reply:
This may be a question for @zhanglong who added the `mMacFrameCounterSet` in PR #9407. 

My objective in this PR is to have a quick fix to address #10633. 

I assume this can be optimized further, but I leave it to you (and maybe other team members who worked on and know more about the "RCP Restoration" feature) to update this in a future PR. Just letting you know, my initial commit/fix did remove this, but then some of the tests failed, mainly `posix-rcp-restoration.exp`. My hunch is that it may have to do with the test itself rather than the change, but leave it to your team to consider and address.

Reply:
sure.

Reply:
I think we can remove the `mMacFrameCounterSet`, but we can enhance it in the future PR.